src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -1433,6 +1442,8 @@ public int Receive(byte[] buffer, int offset, int size, SocketFlags socketFlags,
             int bytesTransferred;
             errorCode = SocketPal.Receive(_handle, buffer, offset, size, socketFlags, out bytesTransferred);
 
+            errorCode = UpdateReceiveSocketErrorForCleanedUp(errorCode, bytesTransferred);
```

Comment:
Will this ever change from a success error code to non-success, or vice versa?  Assuming not, can it be moved off of the success path into the below if block?

Reply:
Yes, for UDP sockets on Linux, when we abort the operation, bytesReceived returns zero with Success. UpdateReceiveSocketErrorForCleanedUp includes a comment about this.